### DISCUSSION TITLE
The error when install bluecfd 2017-2 on windows server 2008 R2 system

### DISCUSSION THREAD

**Discussion by ymg2007** [blueCFD/Core#233] [Show and tell]
Hi, guys. I am a newbie for blueCFD. I installed bluecfd 2017-2 on windows 10. bluecfd can run nice. But when I try to install bluecfd 2017-2 on my high performance server. The error appeared. This server is using Windows server 2008 R2 system. Here is the screenshot of the error. 

![20190710095236](https://user-images.githubusercontent.com/24608776/60934364-834d9680-a2f8-11e9-8115-773e063e59a8.png)

> 
> Setting environment for OpenFOAM 5.x mingw-w64 Double Precision (of5-64), using MSMPI71 - please wait...
> sed：-e 表达式 #1，字符 61：非法回引用
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行117: sed: No such file or directory
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行117: sed: No such file or directory
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行117: sed: No such file or directory
> bash: uname: No such file or directory
> bash: cat: No such file or directory
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行117: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行117: sed: 未找到命令
> bash: sed: No such file or directory
> bash: onWin: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: which: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: cygpath: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: which: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: cygpath: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: which: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: cygpath: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: which: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: cygpath: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: which: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/wmake/scripts/makeReinterpretExePath:行47: cygpath: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> /home/ofuser/blueCFD/OpenFOAM-5.x/bin/foamCleanPath:行136: sed: 未找到命令
> Environment is now ready. Notes:
>    - You can change between installed versions by running: ofmenu
>    - You can change to other predefined versions by running: ofmenuNew
Please give me some help.

**Comment by wyldckat** [comment]
Hi @ymg2007,

A few questions:

1. What is the installation path that you used for blueCFD-Core?

2. If you start again the blueCFD-Core terminal, then run the following command, what does it give you?

```
echo $PATH
```

I ask these questions because it's possible that if you told the blueCFD-Core installer to install in `C:\Program Files\blueCFD-Core-2017`, that the path is not converted to the old 8.3 naming format, for example: `C:\Progra~1\blueCF~1`

If this is the problem, please try to uninstall blueCFD-Core and then install again but at `C:\blueCFD` or a similar path, so that the folder name is short enough to work as intended.

**Comment by ymg2007** [comment]
Hi,wyldckat; 
Thanks for your reply. 
1.The install path of blueCFD-core is C:\Program Files\blueCFD-Core-2017.
2.The output of echo $PATH is here:

> /home/ofuser/blueCFD/ThirdParty-5.x/platforms/mingw_w64Gcc/gperftools-svn/bin:/home/ofuser/blueCFD/ThirdParty-5.x/platforms/mingw_w64Gcc/gperftools-svn/lib:/home/ofuser/blueCFD/OpenFOAM-5.x/platforms/mingw_w64GccDPInt32Opt/lib/MS-MPI-7.1:/home/ofuser/blueCFD/ThirdParty-5.x/platforms/mingw_w64GccDPInt32/lib/MS-MPI-7.1:/home/ofuser/blueCFD/ThirdParty-5.x/platforms/mingw_w64Gcc/MS-MPI-7.1/lib:/home/ofuser/blueCFD/ofuser-of5/platforms/mingw_w64GccDPInt32Opt/lib:/home/ofuser/blueCFD/site/5.x/platforms/mingw_w64GccDPInt32Opt/lib:/home/ofuser/blueCFD/OpenFOAM-5.x/platforms/mingw_w64GccDPInt32Opt/lib:/home/ofuser/blueCFD/ThirdParty-5.x/platforms/mingw_w64GccDPInt32/lib:/home/ofuser/blueCFD/OpenFOAM-5.x/platforms/mingw_w64GccDPInt32Opt/lib/dummy

I will try to re-install blueCFD to C:\blueCFD.

**Comment by wyldckat** [comment]
OK, that demonstrates that there are several paths missing, which is why it fails. Please let me know if reinstalling to `C:\blueCFD` solved the problem or not.

**Comment by ymg2007** [comment]
Hi,wyldckat
I have reinstalled blueCFD to C:\blueCFD. But the error still here.
Here is the screenshot and the output of echo $PATH
![20190711091016](https://user-images.githubusercontent.com/24608776/61014847-04ba2d00-a3bc-11e9-9aff-d83f41b2a958.png)

**Comment by wyldckat** [comment]
There are at least two possibilities for this problem:

1. It's either due to an unexpected character being used that results in the scripts getting confused, so things go wrong,

2. Or there is an application in your Windows Server installation at `C:\Windows` that is affecting the blueCFD-Core installation.


So, in order to attempt to solve this, please do the following steps:

1. Open the terminal again and run this command:

```
./AddOns/notepad2/Notepad2.exe .bashrc
```

* It it doesn't work, then run this command:

```
explorer .
```
Notice that it's a space then a dot at the end of the command. Then in Windows Explorer, right click on the file `.bashrc` and edit with Notepad2.


2. At the start of the file `.bashrc` file , add this line:

```
export LC_ALL=C
```

3. Save the file and and close Notepad2.

4. Close the current terminal and start a new terminal.

5. If it still has problems, then edit the file `.bashrc` once again and add the following line to the start of the file:

```
set -x
```

6. Save the file and and close Notepad2.

7. Start a new terminal window. It should give a lot of text output.

8. You can right-click on the text window and choose the option "Select All", then right-click again and choose "Copy".

9. Then you can paste the output in your next comment, so that I can see the complete output. This will tell us at which point it starts to break apart.
